balena/models/device.py

```diff
@@ -174,6 +174,19 @@ def __set(
                     }
                 )
             else:
+                if len(uuid_or_id_or_ids) < 6:
+                    raise exceptions.AmbiguousDevice(uuid_or_id_or_ids)
+
+                affected = self.__pine.get({
+                    "resource": "device",
+                    "options": {
+                        "$count": {
+                           "$filter": {"uuid": {"$startswith": uuid_or_id_or_ids}}}
+                        }
```

Comment:
This should be marginally faster :)
```suggestion
                        "$top": 2,
                        "$select": "id",
                        "$filter": {"uuid": {"$startswith": uuid_or_id_or_ids}}}
```